packages/cozy-client/src/store/documents.js

```diff
@@ -188,38 +185,67 @@ export const getCollectionFromSlice = (state = {}, doctype) => {
   return Object.values(state[doctype])
 }
 
-/*
-  This method has been created in order to get a returned object
-  in `data` with the full set on information coming potentially from
+/**
+ * 
+ * This method has been created in order to get a returned object
+   in `data` with the full set on information coming potentially from
   `included`
-
+ 
   This method should be somewhere else. The `document` shall not be
   dealt with included / data and so on.
-
+ 
   This method takes `data` and `included` and merge both sources
   together. It should be always up-to-date. The returned object
   will be as full of information as it can be.
-*/
-export const extractAndMergeDocument = (data, updatedStateWithIncluded) => {
-  const doctype = data[0]._type
+ 
+ * @param {Array<import('../types').CozyClientDocument>} newData - Queries documents 
+ * @param {import("../types").DocumentsStateSlice} updatedStateWithIncluded documents' state 
+ * @returns {import("../types").DocumentsStateSlice} the merged documents state
+ */
+export const extractAndMergeDocument = (newData, updatedStateWithIncluded) => {
+  const doctype = newData[0]._type
 
   if (!doctype) {
-    logger.info('Document without _type', data[0])
+    logger.info('Document without _type', newData[0])
     throw new Error('Document without _type')
   }
-  const sortedData = keyBy(data, properId)
-
-  let mergedData = Object.assign({}, updatedStateWithIncluded)
-  mergedData[doctype] = Object.assign({}, updatedStateWithIncluded[doctype])
 
-  Object.values(sortedData).map(data => {
-    const id = properId(data)
-    if (mergedData[doctype][id]) {
-      mergedData[doctype][id] = merge({}, mergedData[doctype][id], data)
-    } else {
-      mergedData[doctype][id] = data
+  // Index docs by id, for faster retrieval
+  const newDocsMap = new Map(newData.map(doc => [properId(doc), doc]))
+
+  const mergedState = Object.assign({}, updatedStateWithIncluded)
+  // FIXME: this was introduced for a bug in banks, but it does not look useful
+  mergedState[doctype] = Object.assign({}, updatedStateWithIncluded[doctype])
+  if (!mergedState[doctype]) {
+    mergedState[doctype] = {}
+  }
+
+  let haveDocumentsChanged = false
+  const deeppmerge = deepmerge()
```

Comment:
nit: maybe `deepmergeFn` or something else without the "double p" typo would be a better name here

Reply:
Done in [999edf3](https://github.com/cozy/cozy-client/pull/1596/commits/999edf30cac5a875f63f7f9549561b2b6a8babac)